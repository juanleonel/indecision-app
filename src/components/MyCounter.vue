<template>
  <h2>Counter {{ counter }}</h2>
  <h2>Square {{ squareCounter }}</h2>

  <div>
    <button class="btn" @click="counter++">+1</button>
    <button class="btn" @click="counter--">-1</button>
  </div>
</template>

<script lang="ts" setup>
import { useCounter } from './composables/useCounter';
interface Props {
  value: number;
}

const props = defineProps<Props>();

const { counter, squareCounter } = useCounter(5);

// import { computed, ref } from 'vue';

// interface Props {
//   value: number;
// }

// const props = defineProps<Props>();

// const counter = ref(props.value);
// const squareCounter = computed(() => counter.value * counter.value);
</script>

<style>
.btn {
  @apply p-5 bg-blue-500 rounded hover:bg-blue-700 mr-2;
}
</style>
