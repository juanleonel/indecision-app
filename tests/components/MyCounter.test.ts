import { describe, should, test } from 'vitest';
import { mount } from '@vue/test-utils';
import MyCounter from '../../src/components/MyCounter.vue';

describe('My Counter component', () => {
  test('should match snapshop', () => {
    const wrapper = mount(MyCounter, {
      props: {
        value: 5,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
