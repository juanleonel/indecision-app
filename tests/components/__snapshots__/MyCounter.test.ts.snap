// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Counter component > should match snapshop 1`] = `
"<h2>Counter 5</h2>
<h2>Square 25</h2>
<div><button class="btn">+1</button><button class="btn">-1</button></div>"
`;
